homeassistant/components/xiaomi_miio/humidifier.py

```diff
@@ -0,0 +1,382 @@
+        )
+        await self.coordinator.async_request_refresh()
+
+        if result:
+            self._state = False
+            self._skip_update = True
+
+    def translate_humidity(self, humidity):
+        """Translate the target humidity to the first valid step."""
+        return (
+            math.ceil(percentage_to_ranged_value((1, self._humidity_steps), humidity))
+            * 100
+            / self._humidity_steps
+            if 0 < humidity <= 100
+            else None
+        )
+
+
+class XiaomiAirHumidifier(XiaomiGenericHumidifierDevice, HumidifierEntity):
+    """Representation of a Xiaomi Air Humidifier."""
+
+    def __init__(self, name, device, entry, unique_id, coordinator):
+        """Initialize the plug switch."""
+        super().__init__(name, device, entry, unique_id, coordinator)
+        if self._model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            self._available_modes = []
+            self._available_modes = [
+                mode.name
+                for mode in AirhumidifierOperationMode
+                if mode is not AirhumidifierOperationMode.Strong
+            ]
+            self._min_humidity = 30
+            self._max_humidity = 80
+            self._humidity_steps = 10
+        elif self._model in [MODEL_AIRHUMIDIFIER_CA4]:
+            self._available_modes = [
+                mode.name for mode in AirhumidifierMiotOperationMode
+            ]
+            self._min_humidity = 30
+            self._max_humidity = 80
+            self._humidity_steps = 100
+        else:
+            self._available_modes = [
+                mode.name
+                for mode in AirhumidifierOperationMode
+                if mode is not AirhumidifierOperationMode.Auto
+            ]
+            self._min_humidity = 30
+            self._max_humidity = 80
+            self._humidity_steps = 10
+
+        self._state_attrs.update(
+            {attribute: None for attribute in self._available_attributes}
+        )
+        self.coordinator.async_add_listener(self.update)
+
+    @callback
+    def update(self):
+        """Fetch state from the device."""
+        state = self.coordinator.data
+        if not state:
+            return
+        _LOGGER.debug("Got new state: %s", state)
+        self._available = True
+        self._state = state.is_on
+        self._state_attrs.update(
+            {
+                key: self._extract_value_from_attribute(state, value)
+                for key, value in self._available_attributes.items()
+            }
+        )
+
+    @property
+    def mode(self):
+        """Return the current mode."""
+        if self._state:
+            return AirhumidifierOperationMode(self._state_attrs[ATTR_MODE]).name
+
+    @property
+    def target_humidity(self):
+        """Return the target humidity."""
+        if self._state:
+            return (
+                self._state_attrs[ATTR_TARGET_HUMIDITY]
```

Comment:
We shouldn't duplicate information in the extra state attributes that we also set in state properties of entities. The base entity of the platform integration will normally set those as state attributes anyway.